contrib/lib/src/compression/mod.rs

```diff
@@ -0,0 +1,138 @@
+//! `Compression` fairing and `Compressed` responder to automatically and
+//! on demand respectively compressing responses.
+mod context;
+mod fairing;
+mod responder;
+
+pub use self::fairing::Compression;
+pub use self::responder::Compressed;
+
+crate use self::context::Context;
+use rocket::http::hyper::header::{ContentEncoding, Encoding};
+use rocket::http::uncased::UncasedStr;
+use rocket::http::MediaType;
+use rocket::{Request, Response};
+use std::io::Read;
+
+#[cfg(feature = "brotli_compression")]
+use brotli::enc::backward_references::BrotliEncoderMode;
+
+#[cfg(feature = "gzip_compression")]
+use flate2::read::GzEncoder;
+
+crate struct CompressionUtils;
+
+impl CompressionUtils {
+    fn accepts_encoding(request: &Request, encoding: &str) -> bool {
+        request
+            .headers()
+            .get("Accept-Encoding")
+            .flat_map(|accept| accept.split(","))
+            .map(|accept| accept.trim())
+            .any(|accept| accept == encoding)
+    }
+
+    fn already_encoded(response: &Response) -> bool {
+        response.headers().get("Content-Encoding").next().is_some()
+    }
+
+    fn set_body_and_encoding<'r, B: Read + 'r>(
+        response: &mut Response<'r>,
+        body: B,
+        encoding: Encoding,
+    ) {
+        response.set_header(ContentEncoding(vec![encoding]));
+        response.set_streamed_body(body);
+    }
+
+    fn skip_encoding(
+        content_type: &Option<rocket::http::ContentType>,
+        content_type_top: &Option<&UncasedStr>,
+        context: &rocket::State<Context>,
+    ) -> bool {
+        let exceptions = &context.exclusions;
+        exceptions
+            .iter()
+            .filter(|c| match content_type {
+                Some(ref orig_content_type) => match MediaType::parse_flexible(c) {
+                    Some(exc_media_type) => {
+                        if exc_media_type.sub() == "*" {
+                            Some(exc_media_type.top()) == *content_type_top
+                        } else {
+                            exc_media_type == *orig_content_type.media_type()
+                        }
+                    }
+                    None => {
+                        if c.contains("/") {
+                            let split: Vec<&str> = c.split("/").collect();
+
+                            let exc_media_type =
+                                MediaType::new(String::from(split[0]), String::from(split[1]));
+
+                            if split[1] == "*" {
+                                Some(exc_media_type.top()) == *content_type_top
+                            } else {
+                                exc_media_type == *orig_content_type.media_type()
+                            }
+                        } else {
+                            false
+                        }
+                    }
+                },
+                None => false,
+            })
+            .count()
```

Comment:
`.filter(|x| /*condition*/).count() > 0` can be reduced to `.any(|x| /*condition*/)`